xlators/mgmt/glusterd/src/glusterd-volgen.c

```diff
@@ -2800,10 +2800,13 @@ validate_user_xlator_position(dict_t *this, char *key, data_t *value,
 {
     int ret = -1;
     int i = 0;
+    char *data = NULL;
 
     if (!value)
         goto out;
 
+    data = value->data;
```

Comment:
`value->data` is what we get from `data_to_str()`. I think we should call this function here instead of on line 2815, and remove the `value_str` variable (or `data`, whatever you prefer).